src/NuGet.Clients/PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -387,5 +466,65 @@ public IEnumerable<NuGetProject> NuGetProjects
                 return _nugetProjects;
             }
         }
+
+        protected void AddBlockedVersions(NuGetVersion[] blockedVersions)
+        {
+            // add a separator
+            if (blockedVersions.Length > 0)
+            {
+                if (_versions.Count > 0)
+                {
+                    _versions.Add(null);
+                }
+
+                var blockedMessage = Resources.Version_Blocked_Generic;
+
+                if (_projectVersionConstraints.All(e => e.IsPackagesConfig))
+                {
+                    // Use the packages.config specific message.
+                    blockedMessage = Resources.Version_Blocked;
+                }
+
+                _versions.Add(new DisplayVersion(new VersionRange(new NuGetVersion(0, 0, 0)), blockedMessage, isValidVersion: false));
+            }
+
+            // add all the versions blocked to disable the update button
+            foreach (var version in blockedVersions)
+            {
+                _versions.Add(new DisplayVersion(version, string.Empty, isValidVersion: false));
+            }
+        }
+
+        protected void SetAutoReferencedCheck(NuGetVersion installedVersion)
+        {
+            var autoReferenced = installedVersion != null
+                    && _projectVersionConstraints.Any(e => e.IsAutoReferenced);
+
+            SetAutoReferencedCheck(autoReferenced);
+        }
+
+        protected void SetAutoReferencedCheck(bool autoReferenced)
+        {
+            if (autoReferenced)
+            {
+                OptionsBlockedUrl = new Uri("https://go.microsoft.com/fwlink/?linkid=841238");
```

Comment:
this link redirects to : [https://docs.microsoft.com/en-gb/](https://docs.microsoft.com/en-gb/) ..doesn't seem right.